src/plugins/data_source_management/public/components/direct_query_data_sources_components/acceleration_creation/create/utils.tsx

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  ACCELERATION_INDEX_NAME_REGEX,
+  ACCELERATION_S3_URL_REGEX,
+} from '../../../../../framework/constants';
+import {
+  AccelerationIndexType,
+  AccelerationRefreshType,
+  CreateAccelerationForm,
+  FormErrorsType,
+  MaterializedViewQueryType,
+  SkippingIndexRowType,
+} from '../../../../../framework/types';
+
+export const pluralizeTime = (timeWindow: number) => {
+  return timeWindow > 1 ? 's' : '';
+};
+
+export const hasError = (formErrors: FormErrorsType, key?: keyof FormErrorsType) => {
+  if (!key) return Object.values(formErrors).some((e) => !!e.length);
+  return !!formErrors[key]?.length;
+};
+
+export const validateDataSource = (dataSource: string) => {
+  return dataSource.trim().length === 0 ? ['Select a valid data source'] : [];
```

Comment:
i18n is needed for all these user facing text, also there are many duplicated text like 'Select a valid xxx', we can use a common base string